### ISSUE TITLE
Display.php  Undefined offset: 0

### ISSUE THREAD

**Issue by wintstar** [SimpleMachines/SMF#4496]
Last Pull Request from today https://github.com/SimpleMachines/SMF2.1/pull/4484
Testet with Xampp
Forum version: SMF 2.1 Beta 4 (more detailed)
Current SMF version: SMF 2.1 Beta 3
GD version: bundled (2.1.0 compatible) 
MySQLi engine: MariaDB 
MySQLi version: 10.1.29-MariaDB 
Alternative PHP Cache: 5.1.2 
PHP: 7.1.11 (more detailed) 
Server version: Apache/2.4.29 (Win32) OpenSSL/1.0.2l PHP/7.1.11 

If I write a post and click on Submit, I will not return to the written post. Only 2 posts per page are displayed in the thread, although 50 are set in the settings. In the error log I get this error message:

>  Today at 08:50
>  Type of error: Critical
> The database value you're trying to insert does not exist: page_id
> Function: Display
> http://localhost/smfbeta/index.php?topic=4.msg76;topicseen
> File: C:/xampp/htdocs/smfbeta/Sources/Display.php
> Line: 917

>  Today at 08:45
>  Type of error: Undefined
> 8: Undefined offset: 0
> http://localhost/smfbeta/index.php?topic=4.50
> File: C:/xampp/htdocs/smfbeta/Sources/Display.php
> Line: 1003

If I then click on the Topics title in the Recent Posts block, I get this error message:

> The database value you're trying to insert does not exist: page_id

**Comment by Sesquipedalian** [comment]
It sounds to me like you recently pulled in a bunch of code changes on top of an old installation that had not been updated since sometime before Aug 23, 2017, which is when the relevant lines were last changed. You should throw that one away and perform a fresh install.

**Comment by wintstar** [comment]
I deleted all files except Settings.php and added  the files  from the newest pull request. I always test with the newest files.

**Comment by Sesquipedalian** [comment]
That doesn't do anything to update your database, though.

**Comment by wintstar** [comment]
Please look here to see how I update. It goes without saying that I also update the database.

http://www.stephan-frank.de/board/index.php/topic,11.msg27.html#msg27

**Comment by Sesquipedalian** [comment]
We do not support upgrading between beta versions of SMF 2.1. The upgrader is only designed for upgrading between released versions (e.g. 1.1 → 2.0, 2.0 → 2.1). As I said, you need to do a fresh install.

**Comment by wintstar** [comment]
My test forum for the German language file is based on a new installation. After errors have been detected, separate reinstallation always takes place. Only if there are the same mistakes, I report this here.

What is described in the topic on my website is a test to upgrade 2.0* to 2.1.  But as described above, the reported bugs are not from this test forum.

**Comment by Sesquipedalian** [comment]
Okay, then I need more information from you in order to reproduce this bug. I have tested just now on my own machines, and I could not reproduce the problem.

**Comment by wintstar** [comment]
With this pull request https://github.com/SimpleMachines/SMF2.1/pull/4488, I didn't have the error. Then I tried a new installation and had the same mistake as here https://github.com/SimpleMachines/SMF2.1/issues/4483. 
Then I updated my test forum to today's Pull Request.I deleted all files and directories except for Settings. php and then I ran upgrade.php.

**Comment by Sesquipedalian** [comment]
Does the error you are reporting in this issue happen with a new installation?

**Comment by wintstar** [comment]
I have installed a new forum again. Since the error was not there, I compared the files. The only difference were the slashes before the directories in the Settings.php
I can't tell if it's the solution. But after the following changes in the settings. php the bug is fixed.
So far, at least in xampp, the values of the directories in the Settings.php file have been created:
`$sourcedir = 'C:\\xampp\\htdocs\\smfbeta\\Sources';`

Now I changed it as it is in a new installation after the last pull request:
`$sourcedir = 'C:\\xampp\\htdocs\\smfbeta/Sources';`

Note the slash after smfbeta.
